runtime/crab/src/pallets/evm_.rs

```diff
@@ -0,0 +1,77 @@
+// --- crates.io ---
+use evm::{executor::PrecompileOutput, Context, ExitError};
+// --- paritytech ---
+use frame_support::{traits::FindAuthor, ConsensusEngineId};
+use sp_core::{crypto::Public, H160, U256};
+use sp_std::marker::PhantomData;
+// --- darwinia-network ---
+use crate::*;
+use darwinia_evm::{runner::stack::Runner, ConcatAddressMapping, Config, EnsureAddressTruncated};
+use darwinia_evm_precompile_issuing::Issuing;
+use darwinia_evm_precompile_simple::{ECRecover, Identity, Ripemd160, Sha256};
+use darwinia_evm_precompile_transfer::Transfer;
+use dp_evm::{Precompile, PrecompileSet};
+use dvm_ethereum::{
+	account_basic::{DvmAccountBasic, KtonRemainBalance, RingRemainBalance},
+	EthereumBlockHashMapping,
+};
+
+pub struct EthereumFindAuthor<F>(sp_std::marker::PhantomData<F>);
+impl<F: FindAuthor<u32>> FindAuthor<H160> for EthereumFindAuthor<F> {
+	fn find_author<'a, I>(digests: I) -> Option<H160>
+	where
+		I: 'a + IntoIterator<Item = (ConsensusEngineId, &'a [u8])>,
+	{
+		if let Some(author_index) = F::find_author(digests) {
+			let authority_id = Babe::authorities()[author_index as usize].clone();
+			return Some(H160::from_slice(&authority_id.0.to_raw_vec()[4..24]));
+		}
+		None
+	}
+}
+
+pub struct CrabPrecompiles<R>(PhantomData<R>);
+impl<R: darwinia_evm::Config> PrecompileSet for CrabPrecompiles<R> {
+	fn execute(
+		address: H160,
+		input: &[u8],
+		target_gas: Option<u64>,
+		context: &Context,
+	) -> Option<Result<PrecompileOutput, ExitError>> {
+		let addr = |n: u64| -> H160 { H160::from_low_u64_be(n) };
+
+		match address {
+			// Ethereum precompiles
+			_ if address == addr(1) => Some(ECRecover::execute(input, target_gas, context)),
+			_ if address == addr(2) => Some(Sha256::execute(input, target_gas, context)),
+			_ if address == addr(3) => Some(Ripemd160::execute(input, target_gas, context)),
+			_ if address == addr(4) => Some(Identity::execute(input, target_gas, context)),
+			// Darwinia precompiles
+			_ if address == addr(21) => Some(<Transfer<R>>::execute(input, target_gas, context)),
+			_ if address == addr(23) => Some(<Issuing<R>>::execute(input, target_gas, context)),
```

Comment:
No `Issuing` precompile for crab now. Remove it.

Reply:
Why remove?

Security issue?

Reply:
Not a security issue, just the time. It's too early.
This precompile is used in combination with `EthereumIssuing` pallet. It would be better to add it to precompile list when we are ready to add `EthereumIssuing` to the Crab network.

Reply:
Agree and also suggest to remove it, is it still in testing on Pangolin, interface might change later.